lib/beaker-pe/install/pe_utils.rb

```diff
@@ -81,6 +81,68 @@ def sorted_hosts subset = hosts
           special_nodes + real_agents
         end
 
+        # If host or opts has the :use_puppet_ca_cert flag set, then push the master's
+        # ca cert onto the given host at /etc/puppetlabs/puppet/ssl/certs/ca.pem.
+        #
+        # This in turn allows +frictionless_agent_installer_cmd+ to generate
+        # an install which references the cert to verify the master when downloading
+        # resources.
+        def install_ca_cert_on(host, opts)
+          if host[:use_puppet_ca_cert] || opts[:use_puppet_ca_cert]
+            @cert_cache_dir ||= Dir.mktmpdir("master_ca_cert")
+            local_cert_copy = "#{@cert_cache_dir}/ca.pem"
+            step "Copying master ca.pem to agent for secure frictionless install" do
+              ca_pem_dir = '/etc/puppetlabs/puppet/ssl/certs'
+              ca_pem_path = "#{ca_pem_dir}/ca.pem"
+              scp_from(master, ca_pem_path , @cert_cache_dir) unless File.exist?(local_cert_copy)
+              on(host, "mkdir -p #{ca_pem_dir}")
+              scp_to(host, local_cert_copy, ca_pem_dir)
+            end
+          end
+        end
+
+        # Generate the command line string needed to from a frictionless puppet-agent
+        # install on this host in a PE environment.
+        #
+        # @param [Host] host The host to install puppet-agent onto
+        # @param [Hash] opts The full beaker options
+        # @option opts [Boolean] :use_puppet_ca_cert (false) if true the
+        #   command will reference the local puppet ca cert to verify the master
+        #   when obtaining the installation script
+        # @param [String] pe_version The PE version string for capabilities testing
+        # @return [String] of the commands to be executed for the install
+        def frictionless_agent_installer_cmd(host, opts, pe_version)
+          # PE 3.4 introduced the ability to pass in config options to the bash
+          # script in the form of <section>:<key>=<value>
+          frictionless_install_opts = []
+          if host.has_key?('frictionless_options') and !  version_is_less(pe_version, '3.4.0')
```

Comment:
What I'd really like to do is remove that block altogether, but I don't remember what our version cut off is for beaker/beaker-pe being able to stand up <2015.2.0